src/crab/ebpf_domain.cpp

```diff
@@ -174,12 +199,77 @@ bool ebpf_domain_t::operator==(const ebpf_domain_t& other) const {
     return stack == other.stack && m_inv <= other.m_inv && other.m_inv <= m_inv;
 }
 
+// Check whether a given type value is within the range of a given type variable's
+// value.
+bool ebpf_domain_t::variable_has_type(variable_t type_variable, int type) const {
+    crab::interval_t interval = m_inv.eval_interval(type_variable);
+    if (interval.is_top())
+        return true;
```

Comment:
shouldn't be `return false` ?

Reply:
As far as I know, no.  top means all values are possible since we don't have any information that would constrain it.  Remember this PR does not change the meaning of the type variable itself.